addons/payment_redsys/models/payment_provider.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.payment_redsys import const
+
+
+class PaymentProvider(models.Model):
+    _inherit = 'payment.provider'
+
+    code = fields.Selection(
+        selection_add=[('redsys', 'Redsys')],
+        ondelete={'redsys': 'set default'}
+    )
+    redsys_merchant_code = fields.Char(string='Redsys Merchant Code', required_if_provider='redsys')
+    redsys_merchant_terminal = fields.Char(string='Redsys Merchant Terminal')
+    redsys_secret_key = fields.Char(
+        string='Redsys Secret Key',
+        required_if_provider='redsys',
+        groups='base.group_system',
+    )
+    signature_version = fields.Char(string='Signature Version', default='HMAC_SHA256_V1')
```

Comment:
Is there a reason why it is saved as a field even though we only use it to pass it to rendering valies. Why not just pass it directly there, since anyway user does not have any option to change the signature version.